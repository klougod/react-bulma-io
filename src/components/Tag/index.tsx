import React, { ReactNode } from 'react'

interface TagProps {
  color?: string
  className?: string
  text?: ReactNode
}

export const Tag = ({ color, className, text }: TagProps) => (
  <span className={`tag ${color} ${className}`}>{text}</span>
)

Tag.defaultProps = {
  color: '',
  className: '',
  text: ''
}

export default Tag
