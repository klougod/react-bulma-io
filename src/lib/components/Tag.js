import React from 'react'
import PropTypes from 'prop-types'

export const Tag = props => {

  const {
    color,
    className,
    text
  } = props

  return (
    <span className={`tag ${color} ${className}`}>{text}</span>
  )
}

Tag.defaultProps = {
  color: '',
  className: '',
  text: ''
}

Tag.propTypes = {
  color: PropTypes.string,
  text: PropTypes.string,
  className: PropTypes.string
}

export default Tag