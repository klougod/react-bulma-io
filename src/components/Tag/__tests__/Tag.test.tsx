import React from 'react'
import Enzyme from 'enzyme'

import Tag from '../'

describe('Tag component test', function () {
  it('should render without crashing', function () {
    const tagMockProps = {
      color: 'red',
      className: 'tagClass',
      text: 'random text'
    }
    const tagMock = Enzyme.shallow(<Tag {...tagMockProps} />)
    expect(tagMock).toMatchSnapshot()
  })

  it('should render span dom component with given props', function () {
    const tagMockProps = {
      color: 'red',
      className: 'tagClass',
      text: 'random text'
    }
    const tagMock = Enzyme.shallow(<Tag {...tagMockProps} />)
    const expectedComponent = (
      <span className={`tag ${tagMockProps.color} ${tagMockProps.className}`}>
        {tagMockProps.text}
      </span>
    )
    expect(tagMock.contains(expectedComponent)).toBe(true)
  })
})
