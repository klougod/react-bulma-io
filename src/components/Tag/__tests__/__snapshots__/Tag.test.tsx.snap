// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag component test should render without crashing 1`] = `
<span
  className="tag red tagClass"
>
  random text
</span>
`;
